internal/controllers/subnet/actuator.go

```diff
@@ -272,12 +274,168 @@ func (actuator subnetActuator) DeleteResource(ctx context.Context, obj orcObject
+	}
+
+	if err != nil {
+		return progress.WrapError(err)
+	}
+
+	return progress.NeedsRefresh()
+}
+
+func handleNameUpdate(updateOpts *subnets.UpdateOpts, obj orcObjectPT, osResource *osResourceT) {
+	name := getResourceName(obj)
+	if osResource.Name != name {
+		updateOpts.Name = &name
+	}
+}
+
+func handleDescriptionUpdate(updateOpts *subnets.UpdateOpts, resource *resourceSpecT, osResource *osResourceT) {
+	description := string(ptr.Deref(resource.Description, ""))
+	if osResource.Description != description {
+		updateOpts.Description = &description
+	}
+}
+
+func handleAllocationPoolsUpdate(updateOpts *subnets.UpdateOpts, resource *resourceSpecT, osResource *osResourceT) {
+	missingAllocationPool := false
+	allocationPools := make([]subnets.AllocationPool, len(resource.AllocationPools))
+	for i := range resource.AllocationPools {
+		allocationPools[i].Start = string(resource.AllocationPools[i].Start)
+		allocationPools[i].End = string(resource.AllocationPools[i].End)
+
+		found := false
+		for _, pool := range osResource.AllocationPools {
+			if pool.Start == allocationPools[i].Start &&
+				pool.End == allocationPools[i].End {
+				found = true
+				break
+			}
+		}
+		if !found {
+			missingAllocationPool = true
+		}
+	}
+	// If the spec doesn't set allocation pools, we'll get a default one and should not try to update it.
+	if len(resource.AllocationPools) > 0 &&
+		(missingAllocationPool || len(resource.AllocationPools) != len(osResource.AllocationPools)) {
+		updateOpts.AllocationPools = allocationPools
+	}
+}
+
+func handleHostRoutesUpdate(updateOpts *subnets.UpdateOpts, resource *resourceSpecT, osResource *osResourceT) {
+	missingHostRoute := false
+	hostRoutes := make([]subnets.HostRoute, len(resource.HostRoutes))
+	for i := range resource.HostRoutes {
+		hostRoutes[i].DestinationCIDR = string(resource.HostRoutes[i].Destination)
+		hostRoutes[i].NextHop = string(resource.HostRoutes[i].NextHop)
+
+		found := false
+		for _, route := range osResource.HostRoutes {
+			if route.DestinationCIDR == hostRoutes[i].DestinationCIDR &&
+				route.NextHop == hostRoutes[i].NextHop {
+				found = true
+				break
+			}
+		}
+		if !found {
+			missingHostRoute = true
+		}
+	}
+	if missingHostRoute || len(resource.HostRoutes) != len(osResource.HostRoutes) {
+		updateOpts.HostRoutes = &hostRoutes
+	}
+}
+
+func handleDNSNameserversUpdate(updateOpts *subnets.UpdateOpts, resource *resourceSpecT, osResource *osResourceT) {
+	missingNameserver := false
+	nameservers := make([]string, len(resource.DNSNameservers))
+	for i := range resource.DNSNameservers {
+		nameservers[i] = string(resource.DNSNameservers[i])
+		found := false
+		for _, nameserver := range osResource.DNSNameservers {
+			if nameserver == nameservers[i] {
+				found = true
+				break
+			}
+		}
+		if !found {
+			missingNameserver = true
+		}
+	}
+	if missingNameserver || len(resource.DNSNameservers) != len(osResource.DNSNameservers) {
+		updateOpts.DNSNameservers = &nameservers
+	}
+}
```

Comment:
This code doesn't handle duplicates. If the original list contains `[]v1alpha1.IPvAny{"one", "two", "two"}` and the updated version contains `[]string{"one", "two", "three"}`, then no modification is applied

Reply:
Also, I suspect that the order of the nameservers is actually relevant to Neutron. In this specific case, I think that the comparison can be much simpler and just compare one by one by index.

Reply:
The first comment above may apply to `handleHostRoutesUpdate` for example.